uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/absence-period-api_test.js

```diff
@@ -0,0 +1,59 @@
+define([
+    'mocks/data/absence-period-data',
+    'leave-absences/shared/apis/absence-period-api',
+  ],
+  function (mockData) {
+    'use strict'
+
+    describe("AbsencePeriodAPI", function () {
+      var AbsencePeriodAPI, $httpBackend;
+
+      beforeEach(module('leave-absences.apis'));
+
+      beforeEach(inject(function (_AbsencePeriodAPI_, _$httpBackend_) {
+        AbsencePeriodAPI = _AbsencePeriodAPI_;
+        $httpBackend = _$httpBackend_;
+      }));
+
+      it("has expected interface", function () {
+        expect(Object.keys(AbsencePeriodAPI)).toContain("all");
+      });
+
+      describe("all()", function () {
+        var absenceTypePromise, totalAbsencePeriods;
+
+        beforeEach(function () {
+          $httpBackend.whenGET(/action=get&entity=AbsencePeriod/)
+            .respond(mockData.all());
+        })
+
+        beforeEach(function () {
+          totalAbsencePeriods = mockData.all().values.length;
+          absenceTypePromise = AbsencePeriodAPI.all();
+        });
+
+        afterEach(function () {
+          //enforce flush to make calls to httpBackend
+          $httpBackend.flush();
+        });
+
+        it("with expected length", function () {
```

Comment:
this description doesn't mean anything, please give meaningful titles to the `it` blocks
```js
it('returns all the absence periods', function () {

});